src/devices_models/devices/common/add_parameters.jl

```diff
@@ -1,27 +1,27 @@
 
 function include_parameters(
     psi_container::PSIContainer,
-    ts_data::Vector{DeviceTimeSeries},
+    constraints::Vector{DeviceTimeSeriesRangeConstraint},
     param_reference::UpdateRef,
     expression_name::Symbol,
     multiplier::Float64 = 1.0,
 )
     @assert model_has_parameters(psi_container)
     time_steps = model_time_steps(psi_container)
     ## TODO, replace multiplier
-    names = [r.name for r in ts_data]
+    names = [get_name(r) for r in constraints]
```

Comment:
I think this one should work as a generator. It was an array before but it is an improvement.